Modules/posixmodule.c

```diff
@@ -322,29 +323,35 @@ corresponding Unix manual entries for more information on calls.");
 #  if defined(TIOCGWINSZ)
 #    define TERMSIZE_USE_IOCTL
 #  endif
-#endif /* MS_WINDOWS */
+#endif /* MS_WINDOWS && !MS_WINDOWS_NON_DESKTOP */
 
 /* Various compilers have only certain posix functions */
 /* XXX Gosh I wish these were all moved into pyconfig.h */
 #if defined(__WATCOMC__) && !defined(__QNX__)           /* Watcom compiler */
 #  define HAVE_OPENDIR    1
 #  define HAVE_SYSTEM     1
 #  include <process.h>
-#else
-#  ifdef _MSC_VER
-     /* Microsoft compiler */
+#elif defined( _MSC_VER)
+  /* Microsoft compiler */
+#  ifndef MS_WINDOWS_NON_DESKTOP
```

Comment:
For `HAVE_GETPPID` we *could* implement `os.getppid()` using the process snapshot API (added in Windows 8) instead of the old toolhelp API. The process snapshot API is available for `WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)` in Windows 10. In this case it would just be the games API family that won't support `os.getppid()`.

Reply:
I think this should d the job: https://github.com/python/cpython/pull/102256/commits/3be5fa71f14e05427d5ff82e19cda95e306413b2